## Step 1. Import numpy and pandas library

In [81]:
# your code here
import numpy as np
import pandas as pd

## Step 2. Load tanzania house-hold data set

In [82]:
# your code here
data = pd.read_csv('data/householdtz.csv')

The columns are organized as:
- HOUSEHOLD number :house_id
- Number of HOUSEHOLD: households_number 
- Number of eligible women in HH: households_women
- Number of eligible men in HH: households_men
- Number of children 5 and under: households_children
- Region: region
- Type of place of residence: residence_type
- Place of residence: residence_place
- Source of drinking water: drink_water_source
- Time to get to water source: time_water_source
- Type of toilet facility: toilet_type
- Has electricity: electricity
- Has radio: radio
- Has television: tv
- Has refrigerator: refrigerator
- Main floor material: floor_material
- Main wall material: wall_material
- Main roof material: roof_material
- Rooms used for sleeping: sleeping_room
- Has a landline telephone: landline
- Share toilet with other households:share_toilet
- Type of cooking fuel: cooking_fuel_type
- Have bednet for sleeping: bednet
- Anything done to water to make safe to drink: treat_water
- Water usually treated by: boil: water_boil
- Water usually treated by: add bleach/chlorine: water_bleach
- Water usually treated by: strain through a cloth: water_strain
- Water usually treated by: use water filter:water_filter
- Water usually treated by: solar disinfection:water_solar
- Water usually treated by: let it stand and settle:water_settle
- Water usually treated by: other: water_other
- Number of households sharing toilet: house_share_toilet
- Wealth index: wealth_index
- Wealth index factor score (5 decimals): wealth_index_score
- Mainland/Zanzibar: location
- Who provides water at main source: water_provider
- Paraffin lamp:paraffin_lamp
- Iron:iron
- Main source of energy for lighting:lighting_source
- How far is the nearest market place (kilometers):distance_market
- How many meals per day:meals_day
- How far is the nearest health facility:distance_health

## Step 3. Insepect the data-set and answer the following questions.

* How many rows does it contain? How many column?

## Step 4 Statistically describe numerical features 
* What is the avaerage number of  households?
* What is the minimum market walking distance?
* What is the average number of meals per day? 
* What is maximum number of women in households?

In [53]:
data.describe()

Unnamed: 0  house_number  households_number  households_women  \
count  401.000000    401.000000         401.000000        401.000000   
mean   200.000000    212.942643           4.720698          0.932668   
std    115.902977    136.938195           2.402456          0.789275   
min      0.000000      1.000000           1.000000          0.000000   
25%    100.000000    101.000000           3.000000          0.000000   
50%    200.000000    201.000000           4.000000          1.000000   
75%    300.000000    305.000000           6.000000          1.000000   
max    400.000000    624.000000          15.000000          5.000000   

       households_men  households_children  time_water_source  sleeping_room  \
count      401.000000           401.000000         401.000000     401.000000   
mean         0.204489             0.987531          56.526185       1.992519   
std          0.466990             0.975881          67.196874       0.904126   
min          0.000000             0.000000           0.000000       1.000000   
25%          0.000000             0.000000          15.000000       1.000000   
50%          0.000000             1.000000          30.000000       2.000000   
75%          0.000000             2.000000          60.000000       3.000000   
max          3.000000             4.000000         600.000000       5.000000   

       wealth_index_score  distance_market   meals_day  distance_health  
count          401.000000       401.000000  401.000000       401.000000  
mean        -35057.356608        39.810474    2.416459         3.573566  
std          70404.433113        36.484024    0.564472         4.477187  
min        -110000.000000         0.000000    1.000000         0.000000  
25%         -84000.000000        15.000000    2.000000         1.000000  
50%         -59000.000000        35.000000    2.000000         2.000000  
75%         -16000.000000        52.000000    3.000000         4.000000  
max         290000.000000       150.000000    3.000000        36.000000

## Counting categorical values

Use: `.value_counts()`

In [94]:
# e.g
data.location.value_counts()

Mainland rural    329
Mainland urban     68
Unguja              4
Name: location, dtype: int64

## Question 1. What is first three regions with lowest average number of house-holds per family?

In [84]:
data.groupby('region')[['households_number']].mean().sort_values(by='households_number', ascending=1)

households_number
region                           
Kagera                   2.500000
Ruvuma                   3.000000
Singida                  3.250000
Pwani                    3.333333
Mtwara                   4.187500
Tanga                    4.238095
Kilimanjaro              4.407407
Mbeya                    4.428571
Iringa                   4.600000
Dodoma                   4.661972
Arusha                   4.666667
Lindi                    5.000000
Mwanza                   5.625000
Morogoro                 6.000000
Kigoma                   6.625000
Zanzibar North           8.500000
Tabora                  10.000000

## Question 2. How many people in Kigoma live in urban area ?

In [85]:
kg=data[(data.residence_type=="Urban") & (data.region=='Kigoma')]
kg.residence_type.value_counts()

Urban    5
Name: residence_type, dtype: int64

## Question 3. What types  of toilet facility do people use. How many in each type?

In [32]:
data['toilet_type'].value_counts()

Pit latrine - without slab / open pit          266
No facility/bush/field                          72
Pit latrine - with slab                         32
Flush - to pit latrine                          18
Pit latrine - ventilated improved pit (VIP)      8
OTHER                                            2
Flush - to septic tank                           2
99                                               1
Name: toilet_type, dtype: int64

## Question 4. What is the common source of drinking water for people living in Arusha?


In [86]:
arusha= data[data.region=='Arusha'][['drink_water_source']]
arusha.drink_water_source.value_counts()

Public tap/standpipe                                    18
River/dam/lake/ponds/stream/canal/irirgation channel    10
Neighbour tap                                            8
Open public well                                         2
Spring                                                   2
Bottled water                                            1
Protected well in dwelling                               1
Name: drink_water_source, dtype: int64

## Question 5. In which region people travel long distance to reach health facility?

In [87]:
data.groupby('region')[['distance_health']].mean().sort_values(by='distance_health', ascending=0)

distance_health
region                         
Tabora                17.000000
Mtwara                 9.375000
Tanga                  7.357143
Ruvuma                 7.000000
Kagera                 4.500000
Morogoro               4.428571
Dodoma                 3.154930
Arusha                 2.595238
Pwani                  2.333333
Iringa                 2.200000
Kigoma                 1.687500
Mwanza                 1.500000
Singida                1.500000
Kilimanjaro            1.259259
Mbeya                  1.000000
Lindi                  1.000000
Zanzibar North         0.000000

## Question 6.Find number of people who do not treat their drinking water in aech region

In [88]:
data[data['treat_water']=="No"]['region'].value_counts()

Dodoma            168
Arusha             29
Tanga              24
Mtwara             15
Kilimanjaro        12
Kigoma              8
Mbeya               5
Morogoro            4
Mwanza              4
Zanzibar North      3
Pwani               3
Singida             3
Iringa              3
Tabora              2
Lindi               1
Ruvuma              1
Kagera              1
Name: region, dtype: int64

## Let convert categorical values into into

In [78]:
data.electricity.value_counts()

No     377
Yes     22
9        2
Name: electricity, dtype: int64

In [89]:
data['electricity'] = data['electricity'].map({'No': 0,'Yes': 1})
data['tv'] = data['tv'].map({'No': 0,'Yes': 1})

To find percent of people with electricity in both urban and rural area we use:

In [90]:
data.groupby('residence_type')[['electricity']].mean()

electricity
residence_type             
Rural              0.024024
Urban              0.212121

This immediately gives us some insight: overall, one of every five family ($0.212 * 5 = 1$) in urban area has electricity, while it is only one in 40 families in rural area.

## Question 8.Find distribution of family with refrigerator in urban and rural area.

In [91]:
data.refrigerator.value_counts()

No     398
Yes      3
Name: refrigerator, dtype: int64

In [92]:
data['refrigerator'] = data['refrigerator'].map({'No': 0,'Yes': 1})

In [93]:
data.groupby('residence_type')[['refrigerator']].mean()

refrigerator
residence_type              
Rural               0.000000
Urban               0.044118